libanalysis/src/fwd_step_enkf.c

```diff
@@ -59,6 +62,7 @@ struct fwd_step_enkf_data_struct {
   long                       option_flags;
   double                     r2_limit;
   bool                       verbose;
+  int                        num_threads;
```

Comment:
Detecting automatically the number of threads. The functionality can be overridden with the flag NUM_THREADS as before. 

Note the cmake changes. 

1. Now OpenMP is default on, which means that it will be in use for the eclipse grid slice functionality. In ecl_grid.c, function **ecl_grid_init_GRDECL_data** the number of threads is not set, so the default is 1?

2. find_package(OpenMP) is not working properly. The c++ compiler has indeed openmp, but this cmake module is saying that it is not there, and unsetting the OpenMP_CXX_FLAGS, necessary to link with openmp. Therefore this workaround that does not use find_package(OpenMP) and sets the flags directly.